libs/core/langchain_core/runnables/graph.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Dict, List, NamedTuple, Optional, Type, Union
+from uuid import uuid4
+
+from langchain_core.pydantic_v1 import BaseModel
+from langchain_core.runnables.base import Runnable
+from langchain_core.runnables.graph_draw import draw
+
+
+class Edge(NamedTuple):
```

Comment:
Is there a benefit to using named tuple here instead of other data classes?

Reply:
i also prefer data classes over named tuples with frozen=True